### PR TITLE
Typescript Support

### PR DESCRIPTION
**Pull request by BlackFenix2**
I converted the ReactCardFlip component from javascript to typescript. i also had to modify the jest config to handle .ts and .tsx files.

i also bumped the entire package.json to the latest dependencies.

let me know if you have any questions.

### CODE REVIEW

**Comment by BlackFenix2** [comment]
i also removed the need to provide a 'key' attribute to the children of the ReactCardFlip Component. I refactored the getComponent function to find the front and back element by children's array index rather than an attribute.

```ts
const getComponent = (key: 0 | 1) => {
    if (props.children.length !== 2) {
      throw new Error(
        'Component ReactCardFlip requires 2 children to function'
      );
    }
    return props.children[key];
  };
```

so now you don't need to specify 'key' anymore.

```tsx
<ReactCardFlip isFlipped={this.state.isFlipped}>
        <div style={this.props.styles.card}>
          <img
            style={this.props.styles.image}
            src="//static.pexels.com/photos/59523/pexels-photo-59523.jpeg"
          />

          <button onClick={this.handleClick}>Flip Card</button>
        </div>

        <div style={this.props.styles.card}>
          <img
            style={this.props.styles.image}
            src="//img.buzzfeed.com/buzzfeed-static/static/2014-04/enhanced/webdr06/4/16/enhanced-11136-1396643149-13.jpg?no-auto"
          />

          <button onClick={this.handleClick}>Flip Card</button>
        </div>
      </ReactCardFlip>
```

**Comment by AaronCCWong** [comment]
Thanks for working on this! I will review asap.

File: package.json [hunk 1]
```diff
@@ -7,34 +7,28 @@
     "build": "webpack --env production --config webpack.dist.js",
     "build-example": "webpack --env production",
     "build-e2e": "webpack --env production --config webpack.e2e.js",
-    "start": "webpack-dev-server --progress --colors --open",
-    "test": "npm run build-e2e && jest",
+    "start": "yarn build-example && webpack-dev-server --progress --colors --open",
```
Comment (AaronCCWong) [inline package.json]: Lets stick with `npm` here.

Comment (BlackFenix2) [reply package.json]: no problem, ill rewrite the scripts for npm and re-test. i should get this done later today or tomorrow.

Comment (BlackFenix2) [reply package.json]: I reverted the scripts back to npm and removed the yarn.lock file

**Comment by BlackFenix2** [comment]
I reverted the changes to package.json to use npm. all of the tests run fine locally. Let me know if you need anything else.

**Comment by AaronCCWong** [comment]
> I reviewed the changes

Can you fix the files that have merge conflicts? Thanks!

**Comment by BlackFenix2** [comment]
oops, still kinds newb-ish in my Git commands. i fixed all of the merge conflicts.

**Pull request merged**